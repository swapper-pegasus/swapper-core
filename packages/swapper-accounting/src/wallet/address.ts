import * as Wallets from '@swapper-org/swapper-wallets'
import { SUPPORTED_TOKENS, MAP_WALLET_CONSTRUCTORS } from '../constants'

type TokenAddresses = {
    symbol: string,
    addresses: Array<string>
}

async function getTokenAddresses (tokenSymbol: string, phrase: string): Promise<TokenAddresses> {
  console.log(phrase)
  const walletInstance: Wallets.IWallet = new Wallets[MAP_WALLET_CONSTRUCTORS[tokenSymbol]](
    process.env[`${tokenSymbol.toUpperCase()}_NODE`],
    Number(process.env[`${tokenSymbol.toUpperCase()}_NETWORK`]),
    'mobile way service edge man luggage hospital garden dolphin flag never insect' // TODO: Replace with real phrase
  )
  const addresses = await walletInstance.getAddresses()
  return {
    symbol: tokenSymbol,
    addresses: addresses
  }
}

async function getAddresses (phrase: string): Promise<TokenAddresses[]> {
  const addressesPromises: Promise<TokenAddresses>[] =
    SUPPORTED_TOKENS.map((symbol) => {
      return getTokenAddresses(symbol, phrase)
    })

  const addresses: TokenAddresses[] = await Promise.all(
    addressesPromises
  )

  return addresses
}

export { getAddresses, TokenAddresses }
