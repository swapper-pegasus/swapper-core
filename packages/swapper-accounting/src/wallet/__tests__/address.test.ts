import { getAddresses } from '../address'

describe('Addresses', () => {
  it('getAllAddresses', async () => {
    const result = await getAddresses('mobile way service edge man luggage hospital garden dolphin flag never insect')
    expect(result).toMatchSnapshot()
  })
})
