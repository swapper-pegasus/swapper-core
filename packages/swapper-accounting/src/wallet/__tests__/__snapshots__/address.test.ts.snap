// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addresses getAllAddresses 1`] = `
Array [
  Object {
    "addresses": Array [
      "Address 1",
      "Address 2",
    ],
    "symbol": "eth",
  },
]
`;
